terraform/module/roles.tf

```diff
@@ -1,42 +1,41 @@
 resource "aws_iam_role" "analyzer_role" {
   name                 = "${var.lambda_function_prefix}-analyzer_role"
   permissions_boundary = var.permissions_boundary
-  path                 = "/${var.lambda_function_prefix}/"
+  path                 = var.role_path_prefix != null ? var.role_path_prefix : "/${var.lambda_function_prefix}/"
```

Comment:
I'd use a local variable to avoid doing the same != null check for each role (see my comment in the main thread).

Reply:
I agree, it makes much more sense!